medmodels/matching/algorithms/propensity_score.py

```diff
@@ -114,24 +122,31 @@ def run_propensity_score(
         treated_array,
         control_array,
         hyperparam=hyperparam,
-        metric=model,
+        model=model,
     )
 
     # Add propensity score to the original data
-    treated_set["Prop. score"] = treated_prop
-    control_set["Prop. score"] = control_prop
+    treated_set = treated_set.with_columns(pl.Series("prop_score", treated_prop))
+    control_set = control_set.with_columns(pl.Series("prop_score", control_prop))
 
     matched_control = nearest_neighbor(
-        treated_set, control_set, metric="absolute", covariates=["Prop. score"]
+        treated_set,
+        control_set,
+        metric=metric,
+        number_of_neighbors=number_of_neighbors,
+        covariates=["prop_score"],
     )
 
-    matched_control.pop("Prop. score")
-    treated_set.pop("Prop. score")
-    control_set.pop("Prop. score")
+    matched_control = matched_control.drop("prop_score")
+    treated_set = treated_set.drop("prop_score")
+    control_set = control_set.drop("prop_score")
 
     return matched_control
 
 
+Model: TypeAlias = Literal["logit", "dec_tree", "forest"]
+
+
```

Comment:
Static definitions and type definitions (`PROP_MODEL` and `Model`) should go to the top of the module